--mm:arc doesn't support msvc c compiler

### Description

When using msvc c compiler, --mm:arc will generate struct initializer which deson't supportted by compiler.
```nim
import tables

proc main*() =
   const cfg_system = {
      "WT_FONTSIZE": "1",
      "WT_FONTSTYLE": "1",
      "CANCEL_HOTKEY_TIP": "1",
   }.toTable

   const tbl_all = {
      "SYSTEM": cfg_system,
   }.toTable

   for key, val in tbl_all:
      echo key, val

when isMainModule:
   main()
```

When compile with msvc c, the error message:
``D:\Tmp\Temp\nimcache\release\vcc-x64\mma56.nim.c(130): error C2059: syntax error: '}'``

And this is the generated c code which cannot compile:
```c
static const struct {
  NI cap; tyTuple__8ZxhPt1NwfElsJs8zUSV9bg data[0];
} TM__5pMwZL7vIJhNbn5JYMH1OQ_4 = {0 | NIM_STRLIT_FLAG, {}};
```



### Nim Version

Nim Compiler Version 1.9.3 [Windows: amd64]
Compiled at 2023-05-14
Copyright (c) 2006-2023 by Andreas Rumpf

active boot switches: -d:release

### Current Output

_No response_

### Expected Output

_No response_

### Possible Solution

_No response_

### Additional Information

_No response_

--- a/compiler/ccgexprs.nim
+++ b/compiler/ccgexprs.nim
@@ -3377,18 +3377,19 @@ proc genConstSeq(p: BProc, n: PNode, t: PType; isConst: bool; result: var Rope) 
 
 proc genConstSeqV2(p: BProc, n: PNode, t: PType; isConst: bool; result: var Rope) =
   let base = t.skipTypes(abstractInst)[0]
-  var data = rope"{"
-  for i in 0..<n.len:
-    if i > 0: data.addf(",$n", [])
-    genBracedInit(p, n[i], isConst, base, data)
-  data.add("}")
+  var data = rope""
+  if n.len > 0:
+    data.add(", {")
+    for i in 0..<n.len:
+      if i > 0: data.addf(",$n", [])
+      genBracedInit(p, n[i], isConst, base, data)
+    data.add("}")
 
   let payload = getTempName(p.module)
-
   appcg(p.module, cfsStrData,
     "static $5 struct {$n" &
     "  NI cap; $1 data[$2];$n" &
-    "} $3 = {$2 | NIM_STRLIT_FLAG, $4};$n", [
+    "} $3 = {$2 | NIM_STRLIT_FLAG$4};$n", [
     getTypeDesc(p.module, base), n.len, payload, data,
     if isConst: "const" else: ""])
   result.add "{$1, ($2*)&$3}" % [rope(n.len), getSeqPayloadType(p.module, t), payload]

